window-purpose-core.el

```diff
@@ -107,13 +108,14 @@ regexp REGEXP."
   "Return the purpose of buffer BUFFER-OR-NAME, as determined by the
 regexps matched by its name.
 REGEXP-CONF is a hash table mapping name regexps to purposes."
-  (car (remove nil
-               (purpose--iter-hash
-                #'(lambda (regexp purpose)
-                    (purpose--buffer-purpose-name-regexp-1 buffer-or-name
-                                                           regexp
-                                                           purpose))
-                regexp-conf))))
+  (cl-block nil
+    (maphash
+     #'(lambda (regexp purpose)
+         (when (purpose--buffer-purpose-name-regexp-1 buffer-or-name
+                                                      regexp
+                                                      purpose)
+           (cl-return purpose)))
+     regexp-conf)))
```

Comment:
This is fine, but I prefer `catch`-`throw` rather than `cl-block`-`cl-return`, purely because they don't require cl-lib. Something like:
```elisp
(catch 'found
  ... (throw 'found purpose) ...)

Reply:
Got it. Replacing the one in `purpose--buffer-purpose-mode` as well